File: backend/app/models/leaderboard.py
from datetime import datetime

from pydantic import BaseModel


class QuizLeaderboardEntryPublic(BaseModel):
    rank: int
    quizId: str
    userId: str | None = None
    nickname: str
    bestScore: int
    lastScore: int
    timesPlayed: int
    updatedAt: datetime


class GlobalLeaderboardEntryPublic(BaseModel):
    rank: int
    userId: str | None = None
    nickname: str
    totalPoints: int
    quizzesPlayed: int
    wins: int
    updatedAt: datetime

File: frontend/src/types/leaderboard.ts
export type QuizLeaderboardEntry = {
  rank: number
  quizId: string
  userId: string | null
  nickname: string
  bestScore: number
  lastScore: number
  timesPlayed: number
  updatedAt: string
}

export type GlobalLeaderboardEntry = {
  rank: number
  userId: string | null
  nickname: string
  totalPoints: number
  quizzesPlayed: number
  wins: number
  updatedAt: string
}

